documentation/src/docs/security/securing-distributed-mode-cluster.md

```diff
@@ -0,0 +1,311 @@
+
+  # Passwords and other arguments entered interactively on the prompt.
+  $ keytool -keystore server01.keystore.jks -alias server01 -genkey
+  ```
+* In the samples below, a weak password `changeit` is used everywhere for easier reading. Be sure to replace it with strong and separate passwords for each file.
+
+### Stage 1: Setting up a Certificate Authority (CA)
+
+If you are going to use an existing public or internal CA service or certificate and key bundle, you may skip this stage.
+
+In this stage, we'll generate a CA in the form of a public-private key pair and a self-signed signed certificate. Later, we'll use the CA to sign other certificates used in the cluster.
+
+1. Create a certificate and key for use as a CA.
+
+   ```bash
+   # All inputs provided using command line arguments
+   $ openssl req -new -x509 -keyout ca-key -out ca-cert -days <validity> \
+            -subj "<distinguished_name>" \
+            -passout pass:<strong_password>
+
+   # Sample command
+   $ openssl req -new -x509 -keyout ca-key -out ca-cert -days 365 \
+            -subj "/C=US/ST=Washington/L=Seattle/O=Pravega/OU=CA/CN=Pravega-CA" \
+            -passout pass:changeit
+   ```
+
+2. Create a truststore containing the CA's certificate.
+
+   This truststore will be used by external and internal clients. External clients are client applications connected to a Pravega cluster. Internal clients are services on the server playing the role of a client in internal communications.
+
+   ```
+   $ keytool -keystore client.truststore.jks -noprompt -alias CARoot -import -file ca-cert \
+        -storepass changeit
+
+   # Optionally, list the truststore's contents to verify everything is in order. The output should show
+   # a single entry with alias name `caroot` and entry type `trustedCertEntry`.
+   $ keytool -list -v -keystore client.truststore.jks -storepass changeit
+   ```
+
+That's it! You now have the following CA and client truststore artifacts:
+
+| File | Description |
+|:-----:|:--------|
+| ca-cert | A file containing CA's certificate in .pem format|
+| ca-key |  A password-protected file containing CA's private key  |
+| client.truststore.jks | A password-protected truststore file containing the CA's certificate |
+
+### Stage 2: Obtaining Server Certificates and keys
+
+This phase is doing the following for each service:
+
+1. Generating server certificates and keys
+2. Generating a Certificate Signing Request (CSR)
+3. Submitting the CSR to a CA and obtain a signed certificate
+4. Preparing a keystore containing the signed server certificate and the CA's certificate
+5. Exporting the server certificate's private key
+
+Note:
+
+* For services running on the same host, you may use the same certificate if those services are exposed using the same hostname/IP address. You may even use wildcard certificates to share certificates across hosts. However, it is strongly recommended that you use seperate certificates for each service.
```

Comment:
For services running on the same host,  same certificates can be used if those services are exposed using the same hostname/IP address.  Wildcard certificates can be used to share certificates across hosts. However, it is strongly recommended, separate certificates should be used  for each service.

Reply:
Done.